docs/upgrade/read-model-for-product-lists.md

```diff
@@ -0,0 +1,140 @@
+# Read Model for Product Lists
+
+There is a new layer in Shopsys Framework, called [read model](/docs/model/introduction-to-read-model.md), separating the templates and application model.
+Besides better logical separation of the application, it is a first step towards usage of Elasticsearch for frontend product lists, and hence significant performance boost in the near future.
+The read model package is marked as *experimental* at the moment so there is a possibility we might introduce some BC breaking changes there.
+You do not need to perform the upgrade instantly, however, if you do so, you will be better prepared for the upcoming changes.
+
+<!-- TODO change link to PR to the split merge commit in project-base -->
+To start using the read model, follow the instructions (you can also find inspiration in [#1018](https://github.com/shopsys/shopsys/pull/1018) where the read model was introduced to `project-base`):
+- add dependency on `shopsys/read-model` to your `composer.json`
+- register the bundle in your `app/AppKernel.php`:
+    ```php
+    // ...
+    class AppKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            $bundles = [
+                // ...
+                new Shopsys\ReadModelBundle\ShopsysReadModelBundle(),
+                // ...
+            ];
+
+            // ...
+
+            return $bundles;
+        }
+
+        // ...
+    }
+    ```
+    - Note: the bundle needs to be registered after `ShopsysFrameworkBundle` as it overwrites its `image` Twig function
+- in all frontend controllers where you are listing products, add new `$listedProductViewFacade` property and modify the constructor to inject instance of `Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFacadeInterface` to the property:
```

Comment:
```suggestion
- in all frontend controllers where you are listing products, add new `$listedProductViewFacade` property and modify the constructor to inject instance of `Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFacadeInterface` to the property. Eg.
```diff
class ProductController extends FrontBaseController
{
+     /**
+      * @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFacadeInterface
+      */
+     private $listedProductsFacade;
```

Please just add here an example, don't apply this suggestion as it would be badly applied probably

Reply:
And the rest should be in another bullet

* And in involved controllers use the facade for getting views
  * in ProductController::detailAction: ...